crates/oxc_ecmascript/src/to_int_32.rs

```diff
@@ -2,79 +2,131 @@
 ///
 /// [ToInt32]: https://tc39.es/ecma262/#sec-toint32
 ///
-/// This is copied from [Boa](https://github.com/boa-dev/boa/blob/61567687cf4bfeca6bd548c3e72b6965e74b2461/core/engine/src/builtins/number/conversions.rs)
+/// This is copied from [Boa](https://github.com/boa-dev/boa/blob/95c8d4820ad10ce32892dd75673b1d8b8854f974/core/engine/src/builtins/number/conversions.rs)
 pub trait ToInt32 {
     fn to_int_32(&self) -> i32;
 }
 
 impl ToInt32 for f64 {
-    #[expect(clippy::float_cmp, clippy::cast_possible_truncation, clippy::cast_possible_wrap)]
     fn to_int_32(&self) -> i32 {
-        const SIGN_MASK: u64 = 0x8000_0000_0000_0000;
-        const EXPONENT_MASK: u64 = 0x7FF0_0000_0000_0000;
-        const SIGNIFICAND_MASK: u64 = 0x000F_FFFF_FFFF_FFFF;
-        const HIDDEN_BIT: u64 = 0x0010_0000_0000_0000;
-        const PHYSICAL_SIGNIFICAND_SIZE: i32 = 52; // Excludes the hidden bit.
-        const SIGNIFICAND_SIZE: i32 = 53;
-
-        const EXPONENT_BIAS: i32 = 0x3FF + PHYSICAL_SIGNIFICAND_SIZE;
-        const DENORMAL_EXPONENT: i32 = -EXPONENT_BIAS + 1;
-
-        fn is_denormal(number: f64) -> bool {
-            (number.to_bits() & EXPONENT_MASK) == 0
+        #[cfg(target_arch = "aarch64")]
+        {
+            f64_to_int32_arm64(*self)
         }
+        #[cfg(not(target_arch = "aarch64"))]
+        {
+            f64_to_int32_generic(*self)
+        }
+    }
+}
 
-        fn exponent(number: f64) -> i32 {
-            if is_denormal(number) {
-                return DENORMAL_EXPONENT;
-            }
+/// Converts a 64-bit floating point number to an `i32` using [`FJCVTZS`][FJCVTZS] instruction on ARM64.
+///
+/// [FJCVTZS]: https://developer.arm.com/documentation/dui0801/h/A64-Floating-point-Instructions/FJCVTZS
+#[cfg(target_arch = "aarch64")]
+fn f64_to_int32_arm64(number: f64) -> i32 {
+    if number.is_nan() {
+        return 0;
+    }
+    let ret: i32;
+    // SAFETY: Number is not nan so no floating-point exception should throw.
+    unsafe {
+        std::arch::asm!(
```

Comment:
The `FJCVTZS` is an instruction available in Arm v8.3-A chips and later. And OXC now is support Arm v8-A: https://github.com/oxc-project/oxc/issues/11045
To avoid the `Illegal instruction` error, we need to detect if `fjcvtzs` is available before use it.